### PR TITLE
Update OpenShift supported versions

### PR DESCRIPTION
**Pull request by pebrc**
In preparation of next release update supported version based on currently release K8s and OCP versions. For Tanzu see #6351

### CODE REVIEW

**Comment by pebrc** [comment]
buildkite test this -f p=ocp,TESTS_MATCH=TestSmoke -m s=8.8.0

**Comment by pebrc** [comment]
buildkite test this -f p=ocp,TESTS_MATCH=TestSmoke,DEPLOYER_CLIENT_VERSION=4.13.4 -m s=8.8.0

**Review by thbkrkr — APPROVED** [review]

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@
     #- DEPLOYER_CLIENT_VERSION: "4.9.58"
     #- DEPLOYER_CLIENT_VERSION: "4.10.55"
     #- DEPLOYER_CLIENT_VERSION: "4.11.33"
-    - DEPLOYER_CLIENT_VERSION: "4.12.14"
+    - DEPLOYER_CLIENT_VERSION: "4.13.4"
```
Comment (thbkrkr) [inline .buildkite/e2e/nightly-main-matrix.yaml:55]: I don't like this duplication but we also maintain the latest version in `hack/deployer/config/plans.yml`.

**Review by barkbay — APPROVED** [review]
LGTM

File: .buildkite/e2e/nightly-main-matrix.yaml (hunk 1 above)
Comment (pebrc) [reply .buildkite/e2e/nightly-main-matrix.yaml:55]: > We have identified an issue that caused pull requests to not reflect additional git pushes over the last several hours. New pushes will now be processed as normal.

The commit that changed the plans.yml should show up now on GH 🤔

**Pull request merged**